File: src/main/java/ru/urstannightmare/cathelpserver/task/DefaultTaskService.java
package ru.urstannightmare.cathelpserver.task;

import ru.urstannightmare.cathelpserver.task.dto.PostTaskRequest;
import ru.urstannightmare.cathelpserver.task.dto.TasksResponse;
import ru.urstannightmare.cathelpserver.task.model.Task;

import java.time.LocalDate;
import java.util.Calendar;

public interface DefaultTaskService {
    TasksResponse getTasksByDate(LocalDate date);
    Task addTask(PostTaskRequest taskDto);
    Task updateDoneState(long id);
    String deleteTask(long id);
    int countAdditionalDaysInPreviousMonth(int dayOfWeek);
    int countAdditionalDaysInNextMonth(int dayOfWeek);

    void resetCalendar(Calendar calendar, LocalDate date);
}

File: src/main/kotlin/ru/urstannightmare/cathelpserver/task/v2/DefaultTaskService.kt
package ru.urstannightmare.cathelpserver.task.v2

import ru.urstannightmare.cathelpserver.task.v2.dto.PostTaskRequest
import ru.urstannightmare.cathelpserver.task.v2.dto.TasksResponse
import java.time.LocalDate

interface DefaultTaskService {
    fun getTasksByDate(date: LocalDate): TasksResponse
    fun addTask(taskDto: PostTaskRequest): TasksResponse
    fun updateDoneState(id: Long): Int
    fun deleteTask(id: Long): String
}